### PR TITLE
SpanContext add deprecated **ID methods

### PR DESCRIPTION
**Pull request by pavolloffay**
Related too https://github.com/uber/jaeger-client-java/pull/149#issuecomment-298712037

### CODE REVIEW

**Comment by pavolloffay** [comment]
crossdock is passing locally. Could you please restart CI?

File: jaeger-core/src/main/java/com/uber/jaeger/SpanContext.java [hunk 1]
```diff
@@ -177,4 +177,44 @@ public static SpanContext withDebugId(String debugId) {
...
+   */
+  @Deprecated
+  public long getParentID() {
+    return parentId;
+  }
+
+  /**
+   * @deprecated use {@link SpanContext#getDebugId()} instead.
+   */
+  @Deprecated
```
Comment (yurishkuro) [inline jaeger-core/src/main/java/com/uber/jaeger/SpanContext.java]: these last two are not needed, external dependencies are only on the trace/span id related fields

**Comment by pavolloffay** [comment]
PR updated.

**Comment by pavolloffay** [comment]
CI failed on a weird error.

**Comment by adriancole** [comment]
So what impact does this have to folks using B3? Is there another way to
get the parent id? Or are you dropping support for B3? cc @felixbarny

**Comment by yurishkuro** [comment]
@adriancole no impact on B3. The non-deprecated methods just have been renamed to ID->Id per Google style. I asked Pavol to add the xxID methods back for backwards compatibility.

**Comment by adriancole** [comment]
cool. thanks for clarifying

**Pull request merged**